src/render/direct3d/SDL_render_d3d.c

```diff
@@ -1020,6 +1020,58 @@ D3D_QueueCopyEx(SDL_Renderer * renderer, SDL_RenderCommand *cmd, SDL_Texture * t
     return 0;
 }
 
+static int
+D3D_QueueGeometry(SDL_Renderer *renderer, SDL_RenderCommand *cmd, SDL_Texture *texture,
+        const float *xy, int xy_stride, const int *color, int color_stride, const float *uv, int uv_stride,
+        int num_vertices, const void *indices, int num_indices, int size_indice,
+        float scale_x, float scale_y)
+{
+    int i;
+    int count = indices ? num_indices : num_vertices;
+    Vertex *verts = (Vertex *) SDL_AllocateRenderVertices(renderer, count * sizeof (Vertex), 0, &cmd->data.draw.first);
+
+    if (!verts) {
+        return -1;
+    }
+
+    cmd->data.draw.count = count;
+
+    for (i = 0; i < count; i++) {
+        int j;
+        float *xy_;
+        SDL_Color col_;
+        if (size_indice == 4) {
```

Comment:
I think it'd be better to pass indices directly into graphics API backends instead of doing CPU-side remapping. Every GPU backend supports hardware-accelerated indexed triangle rendering. There a couple considerations though:
* uint8 indices aren't supported by most modern graphics APIs. Personally I wouldn't bother supporting them.
* uint32 indices aren't supported by some GLES devices. They're still useful to support natively but you might need a fallback like this, just for the GLES backend (I don't know whether D3D9 and D3D11 have a feature query for that, too).

Reply:
Thanks for the infos about int8/32
Actually: librocket uses 'int' and imgui 'short'. Hence this size parameter.
(And I just added the extra uint8 possibility just in case ...) 

I agree, it's better to pass indices directly. But this is more work for this PR, and this remains a possibility for improvement.
(Btw, this is the reason why indice/size/int is sent the back-end and not converted in SDL_render.c)